plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,498 @@
+			client.Connection.Transport = httpTransport(s.InsecureSkipVerify)
+			err := client.requestData(s, sub.Measurement)
+			if err != nil {
+				continue
+			}
+
+			s.Log.Infof("Subscription %v is reconnected", sub.index)
+			return
+		}
+	}()
+}
+
+// addMetric writes sse metric into accumulator
+func (s *ctrlXDataLayerSSE) addMetric(se *sSEEvent, measurement string) {
+	m := metric{}
+	switch se.Event {
+	case "update":
+		// Received an updated value, that we translate into a metric
+		_ = json.Unmarshal([]byte(se.Data), &m)
+		t := convertTimestamp2UnixTime(m.Timestamp)
+		if s.acc != nil {
+			s.wg.Add(1)
+			s.acc.AddFields(measurement, map[string]interface{}{"value": m.Value}, map[string]string{"node": m.Node, "type": m.Type, "host": s.Hostname}, t)
+			s.wg.Done()
+		}
+
+	case "error":
+		// Received an error event, that we report to the accumulator
+		em := errorMetric{}
+		_ = json.Unmarshal([]byte(se.Data), &em)
+		if s.acc != nil {
+			s.wg.Add(1)
+			s.acc.AddError(fmt.Errorf("Received error event for: %v", em.Instance))
+			s.wg.Done()
+		}
+
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		s.Log.Debug("Received keepalive event")
+
+	default:
+		// Received a yet unsupported event type
+		s.Log.Debug("Received unsupported event")
+	}
+}
+
+// Init is for setup, and validating config.
+func (s *ctrlXDataLayerSSE) Init() error {
+	if s.Timeout == 0 {
+		s.Timeout = defaultTimeout
+	}
+	if time.Duration(s.Timeout) < time.Duration(defaultTimeout) {
+		s.Log.Errorf("Setting timeout must be at least %v seconds", time.Duration(defaultTimeout).Seconds())
+		return errors.New("timeout configuration error")
+	}
+
+	for i, sub := range s.Subscription {
+		err := choice.Check(sub.QueueBehaviour, queueBehaviours)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'queue_behaviour' %v is incorrectly configured", i, sub.QueueBehaviour)
+			s.Log.Infof("The right queue behaviour values are %v", queueBehaviours)
+			return err
+		}
+		err = choice.Check(sub.ValueChange, valueChanges)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'value_change' %v is incorrectly configured", i, sub.ValueChange)
+			s.Log.Infof("The right value change values are %v", valueChanges)
+			return err
+		}
+	}
+
+	s.connection = &http.Client{
+		Timeout: time.Duration(s.Timeout),
+	}
+
+	return nil
+}
+
+// Start input as service, retain the accumulator, establish the connection
+// The reconnect will start if the connection is not established
+func (s *ctrlXDataLayerSSE) Start(acc telegraf.Accumulator) error {
+	s.connection.Transport = httpTransport(s.InsecureSkipVerify)
+
+	loginFailed := false
+	if !s.setAuthToken() {
+		s.Log.Errorf("hostname: %v, username: %v log in failed", s.Hostname, s.Username)
+		loginFailed = true
+	}
+
+	s.acc = acc
+
+	for k, l := range s.Subscription {
+		l.index = k
+
+		if !l.Enable {
+			continue
+		}
+
+		if len(l.Nodes) == 0 {
+			continue
+		}
```

Comment:
```suggestion
		if !l.Enable || len(l.Nodes) == 0 {
			continue
		}
```

What is the point in having `enable` here? The user can comment the subscription... Furthermore, I'd rather error out if no nodes are specified or is there a good reason someone would want this?

Reply:
Done (removed)